packages/hawtio/src/plugins/jmx/contexts/Contexts.tsx

```diff
@@ -0,0 +1,135 @@
+import { Card, CardBody, Text } from '@patternfly/react-core'
+import { InfoCircleIcon } from '@patternfly/react-icons'
+import { Table, TableBody, TableHeader, TableProps, wrappable } from '@patternfly/react-table'
+import { IResponse } from 'jolokia.js'
+import { AttributeValues } from '@hawtiosrc/plugins/connect/jolokia-service'
+import { useEffect, useState, useContext } from 'react'
+import { PluginNodeSelectionContext } from '@hawtiosrc/plugins'
+import { log } from '../globals'
+import { contextsService, ContextAttributes } from './contexts-service'
+import { moveElement } from '@hawtiosrc/util/arrays'
+
+export const Contexts: React.FunctionComponent = () => {
```

Comment:
What is the "Context" in JMX plugin?  I'm a bit confused as there's no such concept of "context" in JMX.

Reply:
When the `contexts` node is clicked on within the JMX tree ....

![jmx-contexts-screenshot-old](https://user-images.githubusercontent.com/1634180/221877430-2c205ae2-638d-4b2b-92f0-8f0ef7007634.png)

Old version ^

![jmx-contexts-screenshot-new](https://user-images.githubusercontent.com/1634180/221877454-6ca08203-81d9-4bff-9aa9-ed1117143003.png)

New version ^

Reply:
Its about Camel Contexts and nothing to do with JMX really other than that we grab information via JMX :)

Reply:
So, I started creating it originally as I mistook it for the `expected` contexts page in the camel plugin. Only when I had completed it did I realise that the camel page was indeed different. Therefore, rather than waste it I moved it to the JMX plugin.

I believe the original Attributes components provided this functionality somehow, ie. the rotated table with the attributes as columns, but the react version certainly doesn't. Therefore, for the moment it may be worth keeping just for providing that alternative `view` for any other JMX node that perhaps requires it?

Reply:
For the JMX plugin, I don't think we need to provide a specialised view for some specific MBeans such as Camel contexts. I don't think it was by design in the old Hawtio version, probably it was just a collateral side effect from the specific way of implementation. So long as we can provide a specific view at the dedicated plugin (i.e. Camel plugin for Camel Contexts) it should be fine. Rather it should be better to keep JMX plugin as generic and not to bring any assumptions about specific application context.

Reply:
So, indeed the old Hawtio has a certain feature that shows a table view of MBean attributes on the folder with homogeneous  MBeans. We need to implement the same feature in Hawtio.next. But we don't call it "contexts" but just "attributes" if I'm not mistaken.

![image](https://user-images.githubusercontent.com/156692/222043523-ae231254-cae7-4dc8-84ef-5e9e04f60b4c.png)

Reply:
Right. I'll remove the JMX `Contexts` component (in the git history then in case we need similar functionality) and create an issue for the `homogenous attributes` page view. If you could point out the `old` functionality I'd be grateful as I dug around for ages and failed to find it.

Reply:
https://github.com/hawtio/hawtio-next/issues/125